schema/suggested_genes.js

```diff
@@ -0,0 +1,24 @@
+import fetch from "../lib/apis/fetch"
+import { GraphQLObjectType, GraphQLList, GraphQLString } from "graphql"
+
+const SuggestedGeneType = new GraphQLObjectType({
+  name: "SuggestedGenes",
+  fields: {
+    id: { type: GraphQLString },
+    image_url: { type: GraphQLString },
+    name: { type: GraphQLString },
+    _id: { type: GraphQLString },
+  },
+})
+
+const SuggestedGenes = {
+  type: new GraphQLList(SuggestedGeneType),
+  description: "List of curated genes wit custom images",
+  resolve: async () => {
+    const res = await fetch("https://s3.amazonaws.com/eigen-production/json/eigen_categories.json")
```

Comment:
So psyched to finally see async / await arrive in metaphysics 👍

Reply:
I actually suggested not to use `async`/`await` because it didn't allow us to write a one-liner. We tried:

```js
resolve: async () => await(fetch(URL)).body
```

But Prettier automatically strips out the parenthesis which (I think) is important in terms of semantics. So we needed to do:

```js
resolve: async () => {
  const res = await fetch(URL)
  return res.body
}

```

versus:

```js
resolve: () => fetch(URL).then(({ body }) => body)
```

@damassi do you know any other (and simpler) way to do this?

Reply:
I don't know a better way offhand, and I agree with how it turned out too 👍  it's more elegant

Reply:
Nope, not for a one liner, but this code 

```
resolve: async () => {
  const res = await fetch(URL)
  return res.body
}
``` 
will always win out in my book because at any point another dev could drop in and insert a console.log or breakpoint between `const res` and the `return`, making it faster in the end to track down bugs.

Reply:
I agree. I liked the async / await for readability but didn't have strong opinions and was okay with making the small refactor.

Reply:
> another dev could drop in and insert a console.log or breakpoint between const res and the return, making it faster in the end to track down bugs.

This sounds like an objection against any one-liner function calls, not just this case. What makes this case an exception? Why is it ok to use a one-liner in other places but not here?

Reply:
@yuki24 - Heh 😄  I try to encourage stuff to be broken out more verbosely but one-liners are totally fine! I've just found myself trying to unwind them enough to have gotten in the habit of avoiding them.

Reply:
I hear you @damassi - once you start putting too much stuff in a single line it'll be difficult to figure out what is failing on what, specially in Java. In this case, you could actually easily drop a console.log, so I don't see why `.then` shouldn't be used here:

```js
resolve: () => fetch(URL).then(({ body }) => console.log(body) || body)
```

While the chains for example:

```js
return aaa.bbb.ccc.ddd.fff.eee
```

needs to be broken down to:

```js
const val = aaa.bbb.ccc
console.log(val)
val.ddd.fff.eee
```

cc @craigspaeth

Reply:
Just in case... [I'm NOT advocating promises](https://github.com/artsy/force/blob/master/desktop/apps/artsy_in_miami/index.js#L21-L51). There are certain tools for certain situations, and `async`/`await` just didn't satisfy our need in this case.

Reply:
I'll add my 2cents since I got the @ mention.

I'd probably put myself in the extreme camp of async/await should be the tool for 99.9% of async control flow situations—even when it results in less elegant code to the alternative. It's explicit and easier to read/understand, works with if/else/try/catch, bubbles errors by default without the dev constantly having to remember to `promise().catch()` or `if (err) return callback(err)`, etc. etc. ... and less mixing of patterns is less cognitive overhead. Also, the beauty of JS is how ugly it is from decades of backward compatible baggage. I try to let this free up my cognitive burden from writing elegant solutions and rather focus on designing good modules/libs and keeping the language features to just the minimal set of "good parts".

That said, I'm not strongly opinionated on this case. So 👍  feel free to put it on a single line.

Reply:
I disagree, sorry! What we are discussing here is, "is the one-liner good enough here?", not "promises vs `async`/`await` in general". If I were to do other work in the function (which we were actually discussing doing), I would definitely prefer `async`/`await`. However, if you have two concurrent calls, then that's already a good place for promises. So 99.9% is probably too high or it doesn't make sense when a case-by-case decision should be made.

To me `async`/`await` looks too verbose, and it doesn't contribute to readability.

Reply:
One minor out-of-context point I would like add about `async / await` is that it (on the surface) requires less conceptual understanding than Promises. So while this will likely never be an issue here at Artsy, encountering a Promise in the wild means a) understanding what a `Promise` is; b) understanding how things flow and are passed through the chain; and c) how to properly terminate their execution. 

With `async / await` what is happening is immediately apparent just by the language used to describe it: the block is tagged as `async`, and one `await`s the return of some kind of request or process. No conceptual understanding is needed at a glance, which means that contributations can be made with less immediate cognative overhead. As things scale and teams grow things like this matter more and more. `async / await` allows devs of any level or background to hit the ground running.

Reply:
FWIW the code below is syntactically correct and works fine (which I would have chosen if I had known):

```js
resolve: async () => (await fetch(URL)).body
```

Reply:
FWIW I totally like your implementation too Yuki! I'm down with the implicit return `resolve: () => promise()` style and agree that removing syntax helps readability. It's once you string multiple promises together that async/await really shines—for simple things it's less helpful. 

Personally, I'd reach for async/await the moment I'd need to write a `.then` for all the reasons Chris & highlight—but I'll happily accept readable promising chaining code. 👍  Hope I didn't derail this PR thread or come off dogmatic—just offering my own stylistic preference.

If it's something we feel strongly we want to standardize as a team then I wonder if there's a prettier/linter rule we can implement. Otherwise, I'm down to have a mix of styles in the ecosystems.

Thanks for the conversation, I found this thread interesting/helpful!